WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/address/WooShippingEditAddressViewModel.kt

```diff
@@ -265,15 +275,27 @@ class WooShippingEditAddressViewModel @Inject constructor(
         country.mapLatest { country -> getStatesByCountryCode(country.code) }
             .collectLatest { states ->
                 statesState.value = LocationState.Loaded(states)
-                rawState = ""
-                if (states.isNotEmpty()) {
-                    findLocationByCode(selectedState.value.code, statesState.value)
-                        .takeIf { it != Location.EMPTY }
-                        ?.let { selectedState.value = it } ?: run {
+                val stateCode = if (country.value.code == currentAddress.value.country.code) {
+                    currentAddress.value.state.codeOrRaw
+                } else {
+                    ""
+                }
+                when {
+                    states.isNotEmpty() && stateCode.isNotEmpty() -> {
+                        findLocationByCode(stateCode, statesState.value)
+                            .takeIf { it != Location.EMPTY }
+                            ?.let { selectedState.value = it } ?: run {
+                            selectedState.value = states.first()
+                        }
```

Comment:
np suggestion: This code can be more concise. I used `selectedState.value = ...` only once instead of twice.

```suggestion
                        selectedState.value = findLocationByCode(stateCode, statesState.value)
                            .takeIf { it != Location.EMPTY } ?: states.first()
```